include/FreeRTOSIPConfigDefaults.h

```diff
@@ -396,6 +396,7 @@
 #endif
 
 #if ( ipconfigDHCP_FALL_BACK_AUTO_IP != 0 )
+    #undef ipconfigARP_USE_CLASH_DETECTION
```

Comment:
MISRA checkers do not like the use of `#undef`. Is there another solution?

Reply:
Without `#undef`, we'll have a warning saying that this macro was re-defined. Any suggestions?

Reply:
Here are the default definition:
~~~c
#ifndef ipconfigDHCP_FALL_BACK_AUTO_IP
    #define ipconfigDHCP_FALL_BACK_AUTO_IP     0
#endif

#if ( ipconfigDHCP_FALL_BACK_AUTO_IP != 0 )
    #define ipconfigARP_USE_CLASH_DETECTION    1
#endif

#ifndef ipconfigARP_USE_CLASH_DETECTION
    #define ipconfigARP_USE_CLASH_DETECTION    0
#endif
~~~
which means that all you have to do is define `ipconfigDHCP_FALL_BACK_AUTO_IP` in your FreeRTOSIPConfig.h

Reply:
I pressed the green button too quickly.

Have you defined `ipconfigARP_USE_CLASH_DETECTION` in your FreeRTOSIPConfig.h file?

What about testing as follows:
~~~c
#if ( ipconfigDHCP_FALL_BACK_AUTO_IP != 0 )
    #ifndef ipconfigARP_USE_CLASH_DETECTION
        #define ipconfigARP_USE_CLASH_DETECTION    1
    #else
        #if( ipconfigARP_USE_CLASH_DETECTION != 1 )
            #error ipconfigARP_USE_CLASH_DETECTION should be defined as 1 when AUTO_IP is used.
        #endif
    #endif
#endif
~~~
Would that work for you project?